src/js/migrations.js

```diff
@@ -261,6 +262,40 @@ exports.Migrations= {
     }
   },
 
+  forgetFirstPartySnitches: function (badger) {
+    // this needs to happen before unblockIncorrectlyBlockedDomains
```

Comment:
If we need to rerun `unblockIncorrectlyBlockedDomains()`, we should just call that function manually from our new migration.

Reply:
I am not sure `unblockIncorrectlyBlockedDomains()` is the right approach here though. It only looks at blocked domains, so it wouldn't clean up domains that shouldn't be in `action_map` at all (two `snitch_map` entries --> no `snitch_map` entries).

I think it's worth looking into creating a new `snitch_map` object from the old `snitch_map` object and then calling `mergeUserData()` on the new `snitch_map` + old `action_map`. I could be missing something here though as I haven't tried it myself, just a hunch.

Reply:
Correct me if I'm wrong, but you're suggesting that the logic remain mostly the same, but to append at the end `mergeUserData(snitchMap, actionMap)` ? Perhaps I'm misunderstanding how `mergeUserData` works.

Reply:
The logic will be similar but not the same. Right now you're modifying snitch map directly and then relying on `unblockIncorrectlyBlockedDomains()` to fix action map.

I suggest that `unblockIncorrectlyBlockedDomains()` is not a good fit for this task (as it was made to solve a different problem). I further suggest to try copying snitch map to a temporary snitch map (filtering out not-actually-third-party entries), and then using `mergeUserData()` to rebuild action map and snitch map. The underlying `merge()` function should take care of properly rebuilding action and snitch maps. We may need to blank out existing snitch and action maps right before importing, to clear out the entries for trackers that shouldn't be there any more at all.

Something like the following? Take a look at how and where we use `mergeUserData()` and `merge()` now and what they do.

```javascript
let data = {
  snitch_map: correctedSnitchMap,
  action_map: actionMap,
};
snitchMap.updateObject({});
actionMap.updateObject({});
badger.mergeUserData(data);
```